### PR TITLE
Improved error messages

### PR DESCRIPTION
**Pull request by christopherastone**
Made some common error messages more explicit and/or added clearer formatting.

### CODE REVIEW

File: src/nucleus/jdg.ml [hunk 1]
```diff
@@ -669,7 +673,7 @@ let congr_lambda ~loc (EqTy (ctxa, ta1, ta2))
                  (EqTy (ctxb, b1, b2))
                  (EqTerm (ctxe, e1, e2, ty_e)) =
   if not (TT.alpha_equal_ty b1 ty_e)
-  then error ~loc (RuleInputMismatch ("congr-lambda", b1, ty_e))
+  then error ~loc (RuleInputMismatch ("congr-lambda", b1, "???", ty_e, "???"))
```
Comment (SkySkimmer) [inline src/nucleus/jdg.ml]: The left hand side body type, and the type at which the equality on the bodies is happening.
ie the caller said
```
|- A1 =[Type] A2
x : A1 |- B1 =[Type] B2
x : A1 |- e1 =[C] e2
_____________________________________________________
|- lambda x : A1. B1. e1 =[forall x : A1, B1] lambda x : A2. B2. e2
```
but `C` should be `B1`.

File: src/runtime/runtime.ml [hunk 2]
```diff
@@ -401,9 +403,16 @@ and print_tag ?max_level ~penv t lst ppf =
...
      (* prefix tag applied to one argument *)
-     Print.print ppf ?max_level ~at_level:Level.prefix "%t@ %t"
+     (* Although it is reasonable to parse prefix operators as
+        binding very tightly, it can be confusing to see
+            f SOME x instead of f (SOME x).
+        So we print out unary tags, at least, with the
+        same parenthesization as application, i.e.,
+        Level.app and Level.app_right instead of
+        Level.prefix and Level.prefix_arg *) 
+     Print.print ppf ?max_level ~at_level:Level.app "%t@ %t"
```
Comment (SkySkimmer) [inline src/runtime/runtime.ml]: Some isn't prefix though, it would be more like `f !x`.

File: src/nucleus/jdg.ml (hunk 1 above)
Comment (andrejbauer) [inline src/nucleus/jdg.ml]: I am not sure that any error message which says `congr-lambda` is going to be terribly informative.

Comment (SkySkimmer) [inline src/nucleus/jdg.ml]: If someone is using `congr_lambda` directly then they should know about it. If they're using it through a library they should report to the library people.

**Comment by christopherastone** [comment]
Thanks. I fixed the comment and filled in the missing error strings.

**Pull request merged**